src/main/java/org/example/Main.java

```diff
@@ -1,7 +1,197 @@
+            task = tasks[i];
+            switch (task[2].trim()) {
+                case "true" ->
+                        System.out.printf("%s%d. %s%s%s\n", GREEN_BOLD_BRIGHT, i + 1, RED_BOLD_BRIGHT, String.join(",", task), RESET);
+                case "false" ->
+                        System.out.printf("%s%d. %s%s%s\n", GREEN_BOLD_BRIGHT, i + 1, BLUE_BOLD_BRIGHT, String.join(",", task), RESET);
+            }
+        }
+    }
+
+    private static void addTask() {
+        final var sizeTaskData = 3;
+        final int taskNum = tasks.length + 1;
+        final String[] task = new String[sizeTaskData];
+
+        System.out.printf("\n%sAdding new task Number.%d:%s\n", GREEN_BOLD_BRIGHT, taskNum, RESET);
+        System.out.printf("%sWrite the description of the task:%s ", BLUE_BOLD_BRIGHT, RESET);
+        task[0] = scan.nextLine();
+
+        System.out.printf("%sWrite the date when task end -in the format %sYYYY-MM-DD%s:%s ", BLUE_BOLD_BRIGHT, YELLOW_BOLD_BRIGHT, BLUE_BOLD_BRIGHT, RESET);
+        task[1] = " %s".formatted(dateChecker());//because space after comma
+
+        System.out.printf("%sWrite \"%strue%s\" or \"%sfalse%s\" if task is important or not:%s ", BLUE_BOLD_BRIGHT, YELLOW_BRIGHT, BLUE_BOLD_BRIGHT, YELLOW_BOLD_BRIGHT, BLUE_BOLD_BRIGHT, RESET);
+        while (!scan.hasNext("true") && !scan.hasNext("false")) {
+            scan.nextLine();//for cleaning buffer before new try
+            System.out.printf("\n%sInvalid option - put only %strue%s or %sfalse%s:%s ", RED_BOLD_BRIGHT, YELLOW_BOLD_BRIGHT, RED_BOLD_BRIGHT, YELLOW_BOLD_BRIGHT, RED_BOLD_BRIGHT, RESET);
+        }
+        task[2] = " %s".formatted(scan.next());//next only because if somebody write something after trze/false then would catch also
+        scan.nextLine();//cleaning buffer
+        tasks = ArrayUtils.add(tasks, task);
+        System.out.printf("%sTask  Number %d added%s\n", GREEN_BOLD_BRIGHT, taskNum, RESET);
+    }
+
+    //my method for controlling if string is in right format and the date is at least today
+    private static String dateChecker() {
+        final var today = LocalDate.now();
+        String input;
+        LocalDate date;
+
+        do try {
+            input = scan.nextLine();
+            date = LocalDate.parse(input);
+            if (!date.isBefore(today)) {//date must be at least today if so loop end
+                break;
+            }
+            System.out.printf("%sThis date already was!!%s\n", RED_BOLD_BRIGHT, RESET);
+            System.out.printf("\n%sWrite date which is at least %sToday:%s ", BLUE_BOLD_BRIGHT, RED_BOLD_BRIGHT, RESET);
+        } catch (DateTimeParseException e) {
+            System.out.printf("%sInvalid date format!! %s\n", RED_BOLD_BRIGHT, RESET);
+            System.out.printf("\n%sWrite date in the format %sYYY-MM-DD%s:%s ", BLUE_BOLD_BRIGHT, YELLOW_BOLD_BRIGHT, BLUE_BOLD_BRIGHT, RESET);
+        }
+        while (true);
+        return input;
+    }
```

Comment:
IMPROVE: zjednoduseny a prehlednejsi zapis:

```suggestion
private static String loadAndCheckDate() {
  while (true) {
    try {
      final var today = LocalDate.now();
      final var input = scan.nextLine();
      final var date = LocalDate.parse(input);
      if (!date.isBefore(today)) { // date must be at least today if so loop end
        return input;
      }
      System.out.printf(
          "%sThis date already was!!%s\n", RED_BOLD_BRIGHT, RESET);
      System.out.printf("\n%sWrite date which is at least %sToday:%s ",
          BLUE_BOLD_BRIGHT, RED_BOLD_BRIGHT, RESET);
    } catch (DateTimeParseException e) {
      System.out.printf("%sInvalid date format!! %s\n", RED_BOLD_BRIGHT, RESET);
      System.out.printf("\n%sWrite date in the format %sYYY-MM-DD%s:%s ",
          BLUE_BOLD_BRIGHT, YELLOW_BOLD_BRIGHT, BLUE_BOLD_BRIGHT, RESET);
    }
  }
}
```

Reply:
chápu  mít return ve smyčce a  ale mám dotaz: proč mít definici proměných ve smyččce, myslel jsem že je lepší je definovat vždy mimo smyčku aby se při každém volání smyčky netvořili znova ale jenom se do nich uložili nové hodnoty.
chápu že tady se ideálně bude volat jednou maximálně +- třikrát ale i tak , nebo je to je z hlediska paměti a rychlosti jedno, že se proměnné tvoří pokaždé znova?